client/components/auth-form.jsx

```diff
@@ -3,7 +3,12 @@ import React from 'react';
 export default class AuthForm extends React.Component {
   constructor(props) {
     super(props);
-    this.state = { username: '', password: '' };
+    this.state = {
+      username: '',
+      password: '',
+      error: false,
+      errorMessage: ''
```

Comment:
When possible, try to reduce the number of state properties you are using. I recommend `errorMessage: null` for no error and anything besides `null` is an error. That eliminates the `error` state.

Reply:
That makes sense. I definitely didn't see the path of using `errorMessage` for multiple things instead of just the one thing each for it and `error`.

![image](https://user-images.githubusercontent.com/109567499/202324137-d9590162-5386-4049-9957-1beae9e6ef87.png)

![image](https://user-images.githubusercontent.com/109567499/202324173-fd27699f-325e-4750-8760-3792b780d9c2.png)